index.d.ts

```diff
@@ -118,6 +118,10 @@ export interface FastifyDynamicSwaggerOptions extends FastifySwaggerOptions {
    * Overwrite the route schema
    */
   transform?: Function;
+
+  refResolver?: {
+    buildLocalReference: (json: any, baseUri: any, fragment: string, i: number) => string
```

Comment:
```suggestion
    /** Clone the input schema without changing it. Default to `false`. */
    clone?: boolean;
    buildLocalReference: (
      /** The `json` that is being resolved. */
      json: any,
      /** The `baseUri` object of the schema. */
      baseUri: {
        scheme?: string;
        userinfo?: string;
        host?: string;
        port?: number | string;
        path?: string;
        query?: string;
        fragment?: string;
        reference?: string;
        error?: string;
      },
      /** `fragment` is the `$ref` string when the `$ref` is a relative reference. */
      fragment: string,
      /** `i` is a local counter to generate a unique key. */
      i: number
    ) => string;
```

It was missing the clone option, and baseUri is the parse result from `uri-js`.